okr_app/views.py

```diff
@@ -42,3 +45,27 @@ def update(self, request, *args, **kwargs):
             if request.data.get('issuer', None):
                 del request.data['issuer']
         return super().update(request, *args, **kwargs)
+
+class OKRFileViewset(viewsets.GenericViewSet,
+                 mixins.CreateModelMixin,
+                 mixins.DestroyModelMixin,
+                 mixins.RetrieveModelMixin,
+                 mixins.ListModelMixin):
+    """
+    """
+    permission_classes = [IsAuthenticated]
+    serializer_class = OKRFileSerializer
+    queryset = OKRFile.objects.all()
+
+    def get_queryset(self):
+        user = self.request.user
+        okr = None
+        if self.request.GET.get("okr", None):
+            okr = get_object_or_404(OKR, pk=self.request.GET.get("okr", None))
+        elif self.kwargs.get("pk", None):
+            okr = get_object_or_404(OKR, files=self.kwargs['pk'])
+
+        if okr and (IsApplicationAdminUser().has_permission(self.request, self) or IsOKROwner().has_object_permission(user, okr)):
+            return self.queryset.filter(okr_id=okr.id)
+
+        return self.queryset.none()        
```

Comment:
test the case that a regular user uploads files for OKR that doesn't belong to him/her